### DISCUSSION TITLE
We should explain the benefits of sourcemaps, and when they're not available

### DISCUSSION THREAD

**Discussion by jonbell-lot23** [RecordReplay/devtools#2444] [Ideas]
@jaril and I were discussing how @loganfsmyth is working on manual upload of sourcemaps. This is great.

But it made me realise there are different places in the UI we can bring this to light, both for usability reasons and messaging reasons. Here's what I propose we do:

* Add a visual indicator somewhere that shows if a replay has a sourcemap (library? devtools? both?)
* Consider adding messaging in the skeleton loading screen ("now loading sourcemap" versus "no sourcemap available")
* With these items in place, we'll have another entry point for Logan's "manually upload sourcemap?" call to action

Thoughts?

**Comment by loganfsmyth** [comment]
> Add a visual indicator somewhere that shows if a replay has a sourcemap (library? devtools? both?)

The main issue I see with that is that sourcemaps are per-file, so you could have some random file in your page that happened to have sourcemaps, even though none of the scripts you care about actually have them, so treating it as a recording-wide indicator may well be very hit or miss. 

> Consider adding messaging in the skeleton loading screen ("now loading sourcemap" versus "no sourcemap available")

Similarly for this, currently at a protocol level we never tell the client about a file until the sourcemap for that file has also loaded. I'm not sure that's strictly required, but I do think some things make that assumption right now, and "loading sourcemaps" is a thing that happens all throughout a recording initial load, since scripts can load onto a page at any time, so you can easily be viewing a recording, even if all of the sourcemaps have not loaded yet.

> With these items in place, we'll have another entry point for Logan's "manually upload sourcemap?" call to action

What we could potentially do is, when the user opens a file that has a `//# sourceMappingURL=` comment at the end of it, check to see if we were able to load that sourcemap. If we weren't able to, we could show a little notice with a link to our documentation for uploading. I believe Chome Devtools does that as well.

**Reply by jonbell-lot23** [reply]
Thanks for this context! Because this isn't a clear 1:1 relationship, it sounds like I need to think through this some more.

**Comment by jonbell-lot23** [comment]
Hey @loganfsmyth I'm processing our old discussions. Are there any new insights to add to this since it's been a year?
